bindings/cpp/CMakeLists.txt

```
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.12)
+cmake_policy(VERSION 3.12)
+
+project(PortAudio LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+# Todo (multi-generator): Add support for multiple generators like:
+# - {Debug, Release} x {Static, Dynamic} x {MT, MD (Windows only)}
+set(portaudio-cpp-sources)
+add_subdirectory(source/portaudiocpp) # define source-files
+
+add_library(portaudiocpp ${portaudio-cpp-sources})
+add_library(PortAudio::portaudiocpp ALIAS portaudiocpp) # For subdirectory build
+
+# Remove, when PortAudio::portaudio-cpp has it's own CMakeLists.txt
+if(TARGET portaudio)
+    add_library(PortAudio::portaudio ALIAS portaudio) # Todo (responsibility): move to PortAudio's own CMakeLists.txt
+elseif(NOT TARGET PortAudio::portaudio)
+    find_package(portaudio CONFIG REQUIRED) # Import, if not already existing (e.g. in a sub-project)
+
+    if(TARGET portaudio)
+        add_library(PortAudio::portaudio ALIAS portaudio) # Todo (responsibility): move to PortAudio's own CMakeLists.txt
+    endif()
+endif()
+
+target_link_libraries(portaudiocpp PUBLIC PortAudio::portaudio)
+target_include_directories(portaudiocpp PUBLIC
+    $<INSTALL_INTERFACE:include>
```

```suggestion
    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
```
I think GNUInstallDirs needs to be included before this.

---

No, GNUInstallDirs is only required for the GNUInstallDir's variables.

`$<INSTALL_INTERFACE:include>` means, that the includes can be found in an subdirectory of the folder `include` (which is an relative path). Therefore it will be `$install_dir/include`.